verticapy/datasets.py

```diff
@@ -226,9 +226,16 @@ def load_dataset(
     try:
         vdf = vDataFrame(name, cursor, schema=schema)
     except:
+        name = str_column(name)
+        if not(schema):
+            schema = "v_temp_schema"
+            temp = "LOCAL TEMPORARY "
+        else:
+            schema = str_column(schema)
+            temp = ""
         cursor.execute(
-            "CREATE TABLE {}.{}({});".format(
-                str_column(schema), str_column(name), str_create,
+            "CREATE {}TABLE {}({});".format(
+                temp, str_column(schema) + "." + str_column(name) if schema != 'v_temp_schema' else str_column(name), str_create,
```

Comment:
(1) You should add `ON COMMIT PRESERVE` when you create a temp table; otherwise, its data will be removed after a commit. Look at this page for more info: https://www.vertica.com/docs/10.1.x/HTML/Content/Authoring/SQLReferenceManual/Statements/CREATETEMPORARYTABLE.htm
(2) It would be nice to have vDFs pointing to temp tables, but please be advised that ML training functions currently don't accept temp tables as their input. Therefore, it makes sense if you postpone this change until it is supported on Vertica ML side as well. Otherwise, we'll end up with vDFs that don't work properly for some VerticaPy functions.

Reply:
I just corrected it. It will be available in the next PR

Reply:
Agree!